src/index.ts

```diff
@@ -1590,7 +1589,23 @@ export class BigQuery extends common.Service {
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
       options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+
+      // table#getRows uses listTableData endpoint, which is a faster method
+      // to read rows of results.
+
+      job!.getQueryResults(options, (err, rows) => {
```

Comment:
So `getQueryResults` is going to get called continuously until the job is done and all the rows have been fetched. That doesn't sound like something we want here, so we'll probably need to set `autoPaginate` to `false`.

Reply:
I think I pieced this together, I have a feeling we are going to need to manually poll this method until we see the `jobComplete` value set to `true`. If there are 0 rows, we execute the users callback with an empty array and the response object. If a page was returned, then we should call `listTableData`.

@shollyman does that sound right?

Reply:
Yeah.  Typical flow for query processing should look something like this in terms of backend interactions:

1. Call bigquery.jobs.insert() to insert the query job into the system, 
   which returns a job resource (and job reference) for future polling
   operations.
   
1. Call bigquery.jobs.getQueryResults to poll (possibly repeatedly) for
   job completion.  Typically you do this with a nonzero timeoutMs param (which
   moves polling to the server side as a hanging get), and zero maxRows
   param (as we're not consuming row data here).

1. When job is complete, you can consume schema from the getQueryResults,
   and then setup the node equivalent of a row iterator if the getQueryResults
   response indicates rows were returned.  Many queries don't return results, 
  such as DML or DDL operations (UPDATE TABLE, ALTER TABLE, etc).
   
1. Row iteration should use bigquery.tabledata.list to fetch pages of row data. 
  Each page optionally returns a next page token, which is passed to the next 
  invocation to advance the cursor (don't use explicit row offsets) until no further
  page token is provided.